internal/hcs/system.go

```diff
@@ -347,12 +467,17 @@ func (computeSystem *System) PropertiesV2(ctx context.Context, types ...hcsschem
 	if propertiesJSON == "" {
 		return nil, ErrUnexpectedValue
 	}
-	properties := &hcsschema.Properties{}
-	if err := json.Unmarshal([]byte(propertiesJSON), properties); err != nil {
+	hcsProps := &hcsschema.Properties{}
+	if err := json.Unmarshal([]byte(propertiesJSON), hcsProps); err != nil {
 		return nil, makeSystemError(computeSystem, operation, err, nil)
 	}
+	// Copy over stats if we grabbed these in proc
+	if properties.Statistics != nil {
+		hcsProps.Statistics = properties.Statistics
+		hcsProps.Owner = computeSystem.owner
```

Comment:
Why do we need to update `Owner` here?

Reply:
Owner only gets filled in if we query stats. This is handling the case where we queried stats in the shim and we forwarded the rest to HCS. So filling in Owner on the final properties we return

Reply:
Ah, I see. I think my point is moot here due to the refactoring we are discussing on the other thread. I'll resolve.